File: backend/internal/model/Subject.go
package model

import "gorm.io/gorm"

type Subject struct {
	gorm.Model
	Code             string  `json:"code"`
	Name             string  `json:"name"`
	Credits          int     `json:"credits"`
	Hours            int     `json:"hours"`
	Semester         int     `json:"semester"`
	Career           string  `json:"career"`
	Requirements     string  `json:"requirements"`
	Description      string  `json:"description"`
	RequiredRoomType int     `json:"required_room_type"`
	Groups           []Group `gorm:"many2many:group_subjects;"`
}

File: algoritm-service/src/models/subject.rs
use genevo::{operator::prelude::*, prelude::*, random::Rng, types::fmt::Display};
use serde::{Deserialize, Serialize};
use std::fmt;

// ==============================
// Estructura para cursos
// ==============================
#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct Subject {
    #[serde(rename = "ID")]
    pub id: usize,
    pub name: String,
    pub credits: usize,
    pub hours: u32,
    pub semester: u32,
    pub career: String,
    pub requirements: String,
    pub description: String,
    pub required_room_type: usize,
    #[serde(rename = "specialty")]
    pub speciality: usize,
}
